File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesEpisode.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Lightweight TV episode representation used in TV series details.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesEpisode(
        int id,

        String name,

        String overview,

        @JsonProperty("vote_average")
        double voteAverage,

        @JsonProperty("vote_count")
        int voteCount,

        @JsonProperty("air_date")
        String airDate,

        @JsonProperty("episode_number")
        int episodeNumber,

        @JsonProperty("production_code")
        String productionCode,

        Integer runtime,

        @JsonProperty("season_number")
        int seasonNumber,

        @JsonProperty("show_id")
        int showId,

        @JsonProperty("still_path")
        String stillPath
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesEpisode.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Lightweight TV episode representation used in TV series details.
 *
 * @property id TMDB episode id
 * @property name episode name
 * @property overview episode overview
 * @property voteAverage vote average
 * @property voteCount vote count
 * @property airDate episode air date
 * @property episodeNumber episode number
 * @property productionCode production code
 * @property runtime runtime in minutes
 * @property seasonNumber season number
 * @property showId TMDB TV series id
 * @property stillPath still image path
 */
@JvmRecord
data class TvSeriesEpisode(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("name")
    val name: String?,
    @all:JsonProperty("overview")
    val overview: String?,
    @all:JsonProperty("vote_average")
    val voteAverage: Double,
    @all:JsonProperty("vote_count")
    val voteCount: Int,
    @all:JsonProperty("air_date")
    val airDate: String?,
    @all:JsonProperty("episode_number")
    val episodeNumber: Int,
    @all:JsonProperty("production_code")
    val productionCode: String?,
    @all:JsonProperty("runtime")
    val runtime: Int?,
    @all:JsonProperty("season_number")
    val seasonNumber: Int,
    @all:JsonProperty("show_id")
    val showId: Int,
    @all:JsonProperty("still_path")
    val stillPath: String?,
) : TmdbModel
